File: 1684-count-the-number-of-consistent-strings/1684-count-the-number-of-consistent-strings.cpp
class Solution {
public:
    int countConsistentStrings(string allowed, vector<string>& words) {
        int res = 0;
        bool cset[26] = {false};
        for (auto const &c:allowed) cset[c - 'a'] = true;
        for (auto const &str:words) {
            int is_const = true;
            for (auto const &c:str)
                if (!cset[c - 'a']) is_const = false;
            if (is_const) ++res;
        }
        
        return res;
    }
};

File: 1684-count-the-number-of-consistent-strings/1684-count-the-number-of-consistent-strings.rs
impl Solution {
    pub fn count_consistent_strings(allowed: String, words: Vec<String>) -> i32 {
        let mut res = 0;
        let mut cset = vec![false;26];
        
        for c in allowed.chars() {
            cset[(c as u8 - 'a' as u8) as usize] = true;
        }
        
        for word in words {
            let mut is_const = true;
            for c in word.chars() {
                if cset[(c as u8 - 'a' as u8) as usize] == false {
                    is_const = false;
                }
            }
            if is_const { res += 1; }
        }
        
        return res;
    }
}